types/container/hostconfig_windows.go

```diff
@@ -15,9 +15,38 @@ func (n NetworkMode) IsNone() bool {
 	return n == "none"
 }
 
+// IsContainer indicates whether container uses a container network stack.
+// Returns false as windows doesn't support this mode
+func (n NetworkMode) IsContainer() bool {
+	return false
+}
+
+// IsBridge indicates whether container uses the bridge network stack
```

Comment:
A bridge is a very specific network device that "bridges" layer two packets over an interface, creating a single, layer 2 broadcast domain. A `nat` actually translates IP addresses between two separate networks, operating mostly at layer 3. There are protocols that work with one and not the other. For example, standard DHCP might run fine over a bridge but would require a relay or helper to work across a NAT boundary.

It sounds like windows doesn't really have the concept of a `bridge`. It would be better to leave out the concept of bridge than to have something claim it is a bridge but isn't.

I'm not privy to the implications of this behavior here, but it seems unwise to blur these concepts, as they are _not_ at all the same. Even if they are two approaches to solve the same problem in docker, they have different trade offs and behavior.